lib/closure_tree/model.rb

```diff
@@ -77,7 +77,7 @@ def leaves
     end
 
     def depth
-      ancestors.size
+      ancestor_hierarchies.size
```

Comment:
This code actually changes the behavior `ancestor_hierarchies` includes the self and `ancestors` does not. So this increases the result by 1.

Reply:
I don't understand why this change is needed at all.

Reply:
This makes it so you can call depth on a node in a collection without an n+1 query (except it is off by 1).

`Tag.includes(:ancestor_hierarchies).map(&:depth)`

Reply:
@weilandia can you fix it and add a small test  ? i will release a new version.

Reply:
#435